src/library/scala/collection/mutable/BitSet.scala

```diff
@@ -171,6 +182,113 @@ class BitSet(protected[collection] final var elems: Array[Long])
   override def zip[B](that: IterableOnce[B])(implicit @implicitNotFound(collection.BitSet.zipOrdMsg) ev: Ordering[(Int, B)]): SortedSet[(Int, B)] =
     super.zip(that)
 
+  override def addAll(xs: IterableOnce[Int]): this.type = xs match {
+    case bs: collection.BitSet => this |= bs
+    case range: Range =>
+      if (range.nonEmpty) {
+        val start = range.min
+        if (start >= 0) {
+          val end = range.max
+          val endIdx = end >> LogWL
+          ensureCapacity(endIdx)
+
+          if (range.step == 1 || range.step == -1) {
+            val startIdx = start >> LogWL
+            val wordStart = startIdx * BitSetOps.WordLength
+            val wordMask = -1L << (start - wordStart)
+
+            if (endIdx > startIdx) {
+              elems(startIdx) |= wordMask
+              java.util.Arrays.fill(elems, startIdx + 1, endIdx, -1L)
+              elems(endIdx) |= -1L >>> (BitSetOps.WordLength - (end - endIdx * BitSetOps.WordLength) - 1)
+            } else {
+              elems(endIdx) |= (wordMask & (-1L >>> (BitSetOps.WordLength - (end - wordStart) - 1)))
+            }
+            this
+          } else {
+            super.addAll(range)
```

Comment:
Manually iterating through the `Range` would probably be much faster than using the standard `Growable.addAll` implementation (mainly to avoid boxing) but it doesn't have to be part of this PR.

Reply:
I think I'll leave that to a later, the Range#foreach code can't be directly inlined here because it accesses `private[this] val lastElement`